From 90c4e3754bb97d33a3c52f42fd826f31ba20c866 Mon Sep 17 00:00:00 2001
From: MagneFire <dgriet@gmail.com>
Date: Sat, 9 Apr 2022 17:04:50 +0200
Subject: [PATCH] video: mdp3: Continue when the overlay wasn't released. Under
 some, unknown, circumstances MDP failes to, presumably, release an overlay
 this causes for an endless loop. Where it tries to recover the MDP overlay
 but failing as it appears to be not released and not set (some state in
 between perhaps?). MDP will fail at this point to play a 'new' overlay which
 results in a black screen.

It turns out that we can ignore this error and the black screen will no longer occur.
---
 drivers/video/fbdev/msm/mdp3_ctrl.c | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/drivers/video/fbdev/msm/mdp3_ctrl.c b/drivers/video/fbdev/msm/mdp3_ctrl.c
index 009dda395d59..a35133202d5f 100644
--- a/drivers/video/fbdev/msm/mdp3_ctrl.c
+++ b/drivers/video/fbdev/msm/mdp3_ctrl.c
@@ -1386,8 +1386,6 @@ static int mdp3_overlay_set(struct msm_fb_data_type *mfd,
 	format = mdp3_ctrl_get_source_format(req->src.format);
 
 
-	if (mdp3_session->overlay.id != req->id)
-		pr_err("overlay was not released, continue to recover\n");
 	/*
 	 * A change in overlay structure will always come with
 	 * MSMFB_NEW_REQUEST for MDP3
@@ -1497,12 +1495,6 @@ static int mdp3_overlay_play(struct msm_fb_data_type *mfd,
 
 	mutex_lock(&mdp3_session->lock);
 
-	if (mdp3_session->overlay.id == MSMFB_NEW_REQUEST) {
-		pr_err("overlay play without overlay set first\n");
-		mutex_unlock(&mdp3_session->lock);
-		return -EINVAL;
-	}
-
 	if (mdss_fb_is_power_on(mfd))
 		rc = mdp3_overlay_queue_buffer(mfd, req);
 	else
-- 
2.35.1

